packages/medusa-file-minio/src/services/minio.js

```diff
@@ -46,43 +47,113 @@ class MinioService extends AbstractFileService {
     })
   }
 
-  delete(file) {
-    aws.config.setPromisesDependency(null)
-    aws.config.update({
-      accessKeyId: this.accessKeyId_,
-      secretAccessKey: this.secretAccessKey_,
-      endpoint: this.endpoint_,
-      s3ForcePathStyle: this.s3ForcePathStyle_,
-      signatureVersion: this.signatureVersion_,
-    }, true)
+  async delete(file) {
+    this.updateAwsConfig()
 
     const s3 = new aws.S3()
     const params = {
-      Bucket: this.bucket_,
       Key: `${file}`,
     }
 
-    return new Promise((resolve, reject) => {
-      s3.deleteObject(params, (err, data) => {
-        if (err) {
-          reject(err)
-          return
-        }
-        resolve(data)
-      })
-    })
+    return await Promise.all(
```

Comment:
question: does the upload push the files in both buckets?

Reply:
ok I just wanted to be sure because the delete try to delete from both buckets

Reply:
yeah, that's due to the fact that based on they key which delete takes it cannot determine which bucket the file is in. Delete is idempotent so we can just run delete against both buckets and it will remove the file.

It's a bit botched but when minio doesn't support ACL's on the object level it's necessary. It does impose the constraint that you cannot have duplicate filenames across buckets if you want to remove just one file though.